libtransmission/torrent-files.cc

```diff
@@ -332,16 +336,24 @@ namespace
     std::transform(std::begin(in_upper), std::end(in_upper), std::begin(in_upper), [](auto ch) { return toupper(ch); });
     auto const in_upper_sv = in_upper.sv();
 
+#if defined(_WIN32) || defined(_WIN64)
     static auto constexpr ReservedNames = std::array<std::string_view, 22>{
         "AUX"sv,  "CON"sv,  "NUL"sv,  "PRN"sv, //
         "COM1"sv, "COM2"sv, "COM3"sv, "COM4"sv, "COM5"sv, "COM6"sv, "COM7"sv, "COM8"sv, "COM9"sv, //
         "LPT1"sv, "LPT2"sv, "LPT3"sv, "LPT4"sv, "LPT5"sv, "LPT6"sv, "LPT7"sv, "LPT8"sv, "LPT9"sv, //
     };
+#else
```

Comment:
I don't have a Windows machine to test, but I think `.` and `..` mean the same as POSIx there.

Reply:
All trailing dots are removed on Windows, from previous method `appendSanitizedComponent`